src/history.rs

```diff
@@ -0,0 +1,242 @@
+use anyhow::Result;
+use chrono::{DateTime, Local};
+use csv::*;
+use num::BigUint;
+use serde::{Serialize, Serializer};
+use std::fs::{File, OpenOptions};
+use std::path::Path;
+
+#[derive(Debug, Copy, Clone, Serialize)]
+enum Symbol {
+    BTC,
+    DAI,
+}
+
+#[derive(Debug, Copy, Clone, Serialize)]
+enum Position {
+    Buy,
+    Sell,
+}
+
+#[derive(Debug, Clone, Serialize)]
+struct Float(String);
+
+impl From<f64> for Float {
+    fn from(float: f64) -> Self {
+        Float(float.to_string())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Integer(BigUint);
+
+impl From<BigUint> for Integer {
+    fn from(int: BigUint) -> Self {
+        Integer(int)
+    }
+}
+
+impl From<u64> for Integer {
+    fn from(int: u64) -> Self {
+        Integer(BigUint::from(int))
+    }
+}
+
+impl Serialize for Integer {
+    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.0.to_string())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct PeerId(libp2p::PeerId);
+
+impl From<libp2p::PeerId> for PeerId {
+    fn from(peer_id: libp2p::PeerId) -> Self {
+        Self(peer_id)
+    }
+}
+
+impl Serialize for PeerId {
+    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.0.to_string())
+    }
+}
+
+#[cfg(test)]
+impl Default for PeerId {
+    fn default() -> Self {
+        use std::str::FromStr;
+
+        Self(libp2p::PeerId::from_str("QmUJF1AzhjUfDU1ifzkyuHy26SCnNHbPaVHpX1WYxYYgZg").unwrap())
+    }
+}
+
+#[derive(Debug, Copy, Clone, Serialize)]
+enum FinalState {
+    Refunded,
+    Redeemed,
+}
```

Comment:
:question: Do we really need this? Is a trade that doesn't come to fruition worth adding to the history? Sounds like an extra feature, in case you want to keep track of fees wasted.

Reply:
Let's remove.